src/apps/copo_dtol_submission/views.py

```diff
@@ -139,7 +159,17 @@ def add_sample_to_dtol_submission(request):
         type_sub = profile["type"]
         #is_bge_profile = "BGE" in [ x.get("value","") for x in profile.get("associated_type",[]) ]
         sequence_centres = profile.get("sequencing_centre", [])
-        is_sanger_profile = "erga" in type_sub.lower() and settings.SANGER_SEQUENCING_CENTRE in sequence_centres
+        sequence_centre_objects = SequencingCentre.objects.all()
+
+        # Check if approval is required for the assigned sequencing centre in an ERGA profile
+        is_sequencing_centre_approval_required = "erga" in type_sub.lower() and \
+        any(sc_obj.is_approval_required for sc in sequence_centres for sc_obj in sequence_centre_objects if sc_obj.name == sc)
+    
+        #is_sanger_profile = "erga" in type_sub.lower() and settings.SANGER_SEQUENCING_CENTRE in sequence_centres
+        
+        associated_profile_types_objects = AssociatedProfileType.objects.all()
+        is_associated_project_type_approval_required = "erga" in type_sub.lower() and \
+            any(apt_obj.is_approval_required for apt in associated_profiles for apt_obj in associated_profile_types_objects if apt_obj.name == apt.get("value",""))
```

Comment:
can you review it like this one?  Then there is no need to loop through all the assoicated_profile_type_objects.
 is_associated_project_type_approval_required = any(AssociatedProfileType.objects.filter(is_approval_required=True, name=x.get("value","")) for x in associated_profiles).  Same to the approval checking for sequencing centre